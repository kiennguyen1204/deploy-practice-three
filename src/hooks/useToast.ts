import { useState } from 'react';

interface ToastType {
  status: string;
  message: string;
  openPopup: boolean;
}

const useToast = () => {
  const [toast, setToast] = useState<ToastType>({
    status: '',
    message: '',
    openPopup: false
  });

  const showToast = (status: string, message: string) => {
    setToast({
      status,
      message,
      openPopup: true
    });
  };

  const hideToast = () => {
    setToast({
      status: '',
      message: '',
      openPopup: false
    });
  };

  return {
    toast,
    showToast,
    hideToast
  };
};

export { useToast };
