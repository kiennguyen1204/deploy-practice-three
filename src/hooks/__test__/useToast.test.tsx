import { render, act, renderHook } from '@testing-library/react';
import { useToast } from '../useToast';

describe('useToast', () => {
  it('should initialize with correct initial state', () => {
    const { result } = renderHook(() => useToast());

    expect(result.current.toast).toMatchSnapshot();
  });

  it('should set toast with provided status and message when showToast is called', () => {
    let result: ReturnType<typeof useToast> | undefined;
    const TestComponent = () => {
      result = useToast();
      return null;
    };
    render(<TestComponent />);
    if (result) {
      const { showToast } = result;
      act(() => {
        showToast('success', 'This is a success message');
      });
      expect(result.toast.status).toBe('success');
      expect(result.toast.message).toBe('This is a success message');
      expect(result.toast.openPopup).toBe(true);
    }
  });

  it('should hide toast when hideToast is called', () => {
    let result: ReturnType<typeof useToast> | undefined;
    const TestComponent = () => {
      result = useToast();
      return null;
    };
    render(<TestComponent />);
    if (result) {
      const { showToast, hideToast } = result;
      act(() => {
        showToast('error', 'This is an error message');
      });
      expect(result.toast.openPopup).toBe(true);
      act(() => {
        hideToast();
      });
      expect(result.toast.openPopup).toBe(false);
    }
  });
});
