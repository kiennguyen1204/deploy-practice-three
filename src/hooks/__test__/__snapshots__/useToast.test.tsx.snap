// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useToast should initialize with correct initial state 1`] = `
{
  "message": "",
  "openPopup": false,
  "status": "",
}
`;
